### PR TITLE
Fix/3415  publish tlm with 0mq pushpin

### PR DESCRIPTION
**Pull request by grignards**
See #3415.

## Checkpoints

These points must be checked before merging. Please don't edit them out.

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/doc/live_message_setup.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ you need to set tracim config like this to use default pushpin port:
 
 ~~~ini
 basic_setup.website_base_url = http://localhost:7999
-live_messages.control_uri = http://localhost:5561
+live_messages.control_zmq_uri = tcp://localhost:5563
```
Comment (raphj) [inline backend/doc/live_message_setup.md:20]: Would it be possible to use a UNIX socket (maybe more efficient than local tcp?)
Probably far from being a bottleneck though.

File: backend/tracim_backend/tests/functional/test_live_messages.py [hunk 2]
```diff
@@ -1,20 +1,51 @@
...
 from tracim_backend.tests.fixtures import *  # noqa: F403,F40
 
 
+@pytest.fixture
+def big_html_document(workspace_api_factory, content_api_factory, session) -> Content:
+    workspace_api = workspace_api_factory.get()
+    workspace = workspace_api.create_workspace(label="Foobar")
+    content_api = content_api_factory.get()
+
+    description = "".join(random.choice(string.ascii_letters) for _ in range(2000000))
```
Comment (raphj) [inline backend/tracim_backend/tests/functional/test_live_messages.py:23]: I don't think we should have randomness in tests. Is it sufficient to generate a zeroed string, or provide a seed so we are sure the test is always the same / deterministic? Not that it should impact anything here, but still.

**Comment by raphj** [comment]
Don't forget to document the solution in the issue :-)

File: backend/doc/live_message_setup.md (hunk 1 above)
Comment (grignards) [reply backend/doc/live_message_setup.md:20]: maybe, but this involves modifying pushpin's configuration (as it uses tcp by default) so I propose to leave this value for now.

File: backend/tracim_backend/tests/functional/test_live_messages.py (hunk 2 above)
Comment (grignards) [reply backend/tracim_backend/tests/functional/test_live_messages.py:23]: You're right, I'll change this to a fixed content.

File: backend/doc/live_message_setup.md (hunk 1 above)
Comment (raphj) [reply backend/doc/live_message_setup.md:20]: :+1:

File: backend/tracim_backend/tests/functional/test_live_messages.py (hunk 2 above)
Comment (lebouquetin) [reply backend/tracim_backend/tests/functional/test_live_messages.py]: I suggest to also put UFT-8 chars...

**Comment by PhilippeAccorsi** [comment]
A small fix is also added in this MR about missing `apt update` before installing pushpin in docker images.

File: backend/tracim_backend/tests/functional/test_live_messages.py (hunk 2 above)
Comment (raphj) [reply backend/tracim_backend/tests/functional/test_live_messages.py]: After discussion, we are putting this for future work to ease the release of Tracim 3.0.0. Unexpectedly, putting Japanese characters in there makes the test ruuuuunnnn veeerrrryyy sloooowly.

File: backend/tracim_backend/tests/functional/test_live_messages.py [hunk 3]
```diff
@@ -18,7 +18,12 @@ def big_html_document(workspace_api_factory, content_api_factory, session) -> Co
     workspace = workspace_api.create_workspace(label="Foobar")
     content_api = content_api_factory.get()
 
-    description = 2000000 * "a"
+    # NOTE: MySQL and MariaDB have a maximum of 65536 bytes for description,
+    # so the size is chosen accordingly
```
Comment (raphj) [inline backend/tracim_backend/tests/functional/test_live_messages.py:22]: Won't this effectively break Tracim 3.0 on MySQL / MariaDB as soon as someone writes a document bigger than 64 KiB?

Should we prevent people using MySQL from migrating to Tracim 3.0 until we sort this out?

Comment (raphj) [reply backend/tracim_backend/tests/functional/test_live_messages.py:22]: oh, by the way, you should put `2020-08-07 - SG` - in this comment.

Comment (lebouquetin) [reply backend/tracim_backend/tests/functional/test_live_messages.py:22]: > Should we prevent people using MySQL from migrating to Tracim 3.0 until we sort this out?

The problem is already there.

**Pull request merged**